src/t8_geometry/t8_geometry_implementations/t8_geometry_analytic.h

```diff
@@ -79,19 +84,30 @@ T8_EXTERN_C_BEGIN ();
 void
 t8_geometry_analytic_destroy (t8_geometry_c **geom);
 
-/** Create a new analytical geometry.
- * \return          A pointer to an allocated geometry struct.
- */
+/**
+   * Create a new analytic geometry with a given dimension. The geometry
+   * is viable with all tree types and uses a user-provided analytic and
+   * jacobian function. The actual mappings are done by these functions.
+   * \param [in] dim        The dimension of this geometry.
+   * \param [in] name       The name to give this geometry.
+   * \param [in] analytical The analytical function to use for this geometry.
+   * \param [in] jacobian   The jacobian of \a analytical.
+   * \param [in] load_tree_data The function that is used to load a tree's data.
+   * \param [in] tree_negative_volume_in The function that is used to compute if a trees volume is negative.
+   * \param [in] tree_compatible_in The function that is used to check if a tree is compatible with the geometry.
```

Comment:
please add documentation for `const void *user_data`

Reply:
see 65e363a18